### PR TITLE
fix compatibility with PHP 7.4

### PR DESCRIPTION
**Pull request by smoench**

### CODE REVIEW

**Comment by smoench** [comment]
There is still one error with PHP 7.4 which happens several times:
```
1) Mockery_AdhocTest::testMockeryInterfaceForInterface
Cannot use "parent" when current class scope has no parent
/home/travis/build/mockery/mockery/library/Mockery/Loader/EvalLoader.php:34
/home/travis/build/mockery/mockery/library/Mockery/Loader/EvalLoader.php:34
/home/travis/build/mockery/mockery/library/Mockery/Container.php:230
/home/travis/build/mockery/mockery/tests/Mockery/AdhocTest.php:59
```
But I don't where and how to start debugging. Anyone has a hint?

File: library/Mockery.php [hunk 1]
```diff
@@ -858,7 +858,8 @@ private static function getNewDemeterMock(
                 if ($parRefMethodRetType !== null) {
                     $nameBuilder = new MockNameBuilder();
                     $nameBuilder->addPart('\\' . $newMockName);
-                    $mock = self::namedMock($nameBuilder->build(), (string) $parRefMethodRetType);
+                    $type = PHP_VERSION_ID >= 70100 ? $parRefMethodRetType->getName() : (string)$parRefMethodRetType;
```
Comment (GrahamCampbell) [inline library/Mockery.php]: Maybe it's worth having a utility method in this library which does this, instead of copying the logic 4 times?

Comment (smoench) [reply library/Mockery.php]: I think it's not worth it and would be easier to remove them anytime soon when the minimum PHP version gets increased.

**Comment by smoench** [comment]
I could fix the code preventing the `Cannot use "parent" when current class scope has no parent` php deprecation.

**Comment by GrahamCampbell** [comment]
Just tested this with PHP 7.4.0alpha2. Works perfectly. :)

**Comment by franzliedke** [comment]
Good to see this being covered. I also [ran into](https://travis-ci.org/flarum/core/jobs/554884397) the "Cannot use parent..." error.

Thanks for Mockery! :clap:

**Comment by GrahamCampbell** [comment]
Friendly ping @davedevelopment. :)

**Comment by davedevelopment** [comment]
Thank you! Sorry for the delay, my head's a shed at the minute.

**Pull request merged**